admin-dev/themes/default/js/bundle/product/form.js

```diff
@@ -1771,57 +1793,296 @@ var priceCalculation = (function() {
-      priceToBeChanged = truncateDecimals(priceToBeChanged, 6);
+      this.impactFinalPriceTaxExcluded(obj);
+      this.impactFinalPriceTaxIncluded(obj);
+    },
 
-      finalPrice.html(priceToBeChanged);
+    /**
+     * @param {jQuery} obj
+     */
+    impactFinalPriceTaxExcluded(obj) {
+      var combinationForm = obj.closest('div[id^="combination_form_"]');
+
+      var impactPriceTE = this.getImpactTEInputValue(obj);
+      var ecotaxTE = this.getCombinationEcotaxTaxExcluded(obj);
+      // If no ecotax for combination use the product's one
+      if (ecotaxTE <= 0) {
+        ecotaxTE = getEcotaxTaxExcluded();
+      }
+
+      var finalPriceContainer = combinationForm.find('.final-price');
+      var productPriceTE = priceCalculation.getProductBasePrice();
+      var finalPriceTE = productPriceTE + impactPriceTE + ecotaxTE;
+      finalPriceTE = ps_round(finalPriceTE, displayPricePrecision);
+      finalPriceTE = truncateDecimals(finalPriceTE, displayPricePrecision);
+
+      finalPriceContainer.html(finalPriceTE);
+      finalPriceContainer.data('price', finalPriceTE);
+    },
+
+    /**
+     * @param {jQuery} obj
+     */
+    impactFinalPriceTaxIncluded(obj) {
+      var combinationForm = obj.closest('div[id^="combination_form_"]');
+
+      var impactPriceTE = this.getImpactTEInputValue(obj);
+      var ecotaxTI = this.getCombinationEcotaxTaxIncluded(obj);
+      // If no ecotax for combination use the product's one
+      if (ecotaxTI <= 0) {
+        ecotaxTI = getEcotaxTaxIncluded();
+      }
+
+      var finalPriceTIContainer = combinationForm.find('.final-price-tax-included');
+      var productPriceTE = priceCalculation.getProductBasePrice();
+      var finalPriceTI = this.computePriceTaxIncluded(productPriceTE + impactPriceTE) + ecotaxTI;
+      finalPriceTI = ps_round(finalPriceTI, displayPricePrecision);
+      finalPriceTI = truncateDecimals(finalPriceTI, displayPricePrecision);
+
+      finalPriceTIContainer.html(finalPriceTI);
+      finalPriceTIContainer.data('price', finalPriceTI);
+    },
+
+    /**
+     * Calculates the impact on price so that the change on ecotax doesn't affect the final price
+     * @param {jQuery} obj
+     */
+    impactPricesForEcotax: function (obj) {
+      var finalPriceTIContainer = obj.closest('div[id^="combination_form_"]').find('.final-price-tax-included');
+      var currentFinalPriceTI = Number(finalPriceTIContainer.data('price'));
+      var productPrice = priceCalculation.getProductBasePrice();
+      var productPriceTI = this.computePriceTaxIncluded(productPrice);
+      var ecotaxTI = this.getCombinationEcotaxTaxIncluded(obj);
+      // If no ecotax for combination use the product's one
+      if (ecotaxTI <= 0) {
+        ecotaxTI = getEcotaxTaxIncluded();
+      }
+
+      // Compute impact price tax excluded then update the price tax included
+      var impactPriceTI = currentFinalPriceTI - ecotaxTI - productPriceTI;
```

Comment:
I'm soooo afraid of mathematical inaccurate operations 😨  in this statement

Reply:
You mean just like every computing we have so far in all the BO? 😅
(Except ONE use case in new product page)

Reply:
Yes 😛  but making the list bigger is no good

Out of the scope of this PR I agree 👍 

By the way, on Product Page V2, do we have something for accurate JS mathematical operations?

Reply:
Yep now we do we introduced `bignumber.js` That's the ONE use case I mentioned, it's only implemented for product price computing for now, but it should be used everywhere in this page

That's why I don't mind adding some additional float computing since the page is being migrated anyway, and the idea is not to rework the whole legacy page for a bug fix